### DISCUSSION TITLE
Numlock not mapped until keyboard plugged in

### DISCUSSION THREAD

**Discussion by strukiii** [tiny-pilot/tinypilot#1753] [General]
Hi, weird anecdote from setting up the tinypilot on a fresh windows system.  The numpad keys wouldn't work initially, but after plugging in a real keyboard to the target system the numpad maps over fine.  Important to note this is a fresh computer out of the box, where Its never had real peripherals plugged in other than the virtual usb-gadget.  Not sure if the driver is installed on the windows machine after the keyboard is plugged in?  Anyways, Im sure low prio I've fixed it myself but wanted to share.

**Comment by cghague** [comment]
Hi strukiii; thanks for sharing your observations with us!

We've not encountered this behavior before, and based on some quick experiments, toggling the numlock key on a physical keyboard doesn't seem to impact whether the number keys in the TinyPilot virtual keyboard work. Your theory about this being a brand-new computer that needed initializing in some fashion sounds plausible.

I'm pleased to hear that the number keys are working for you now, but if you encounter the issue again in the future, can you send us a link to your TinyPilot logs directly after pressing a number key on the virtual keyboard? You can do this by going to System, then Logs, then Get Shareable URL.
